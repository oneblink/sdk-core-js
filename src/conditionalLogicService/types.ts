import { FormTypes, SubmissionTypes } from '@oneblink/types'

export type FormElementsCtrl = {
  model: SubmissionTypes.S3SubmissionData['submission']
  flattenedElements: FormTypes.FormElement[]
  parentFormElementsCtrl?: FormElementsCtrl
}
