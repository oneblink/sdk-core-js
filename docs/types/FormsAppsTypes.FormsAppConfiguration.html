<!DOCTYPE html><html class="default" lang="en" data-base="../"><head><meta charset="utf-8"/><meta http-equiv="x-ua-compatible" content="IE=edge"/><title>FormsAppConfiguration | @oneblink/sdk-core</title><meta name="description" content="Documentation for @oneblink/sdk-core"/><meta name="viewport" content="width=device-width, initial-scale=1"/><link rel="stylesheet" href="../assets/style.css"/><link rel="stylesheet" href="../assets/highlight.css"/><script defer src="../assets/main.js"></script><script async src="../assets/icons.js" id="tsd-icons-script"></script><script async src="../assets/search.js" id="tsd-search-script"></script><script async src="../assets/navigation.js" id="tsd-nav-script"></script><script async src="../assets/hierarchy.js" id="tsd-hierarchy-script"></script></head><body><script>document.documentElement.dataset.theme = localStorage.getItem("tsd-theme") || "os";document.body.style.display="none";setTimeout(() => window.app?app.showPage():document.body.style.removeProperty("display"),500)</script><header class="tsd-page-toolbar"><div class="tsd-toolbar-contents container"><a href="../index.html" class="title">@oneblink/sdk-core</a><div id="tsd-toolbar-links"></div><button id="tsd-search-trigger" class="tsd-widget" aria-label="Search"><svg width="16" height="16" viewBox="0 0 16 16" fill="none" aria-hidden="true"><use href="../assets/icons.svg#icon-search"></use></svg></button><dialog id="tsd-search" aria-label="Search"><input role="combobox" id="tsd-search-input" aria-controls="tsd-search-results" aria-autocomplete="list" aria-expanded="true" autocapitalize="off" autocomplete="off" placeholder="Search the docs" maxLength="100"/><ul role="listbox" id="tsd-search-results"></ul><div id="tsd-search-status" aria-live="polite" aria-atomic="true"><div>Preparing search index...</div></div></dialog><a href="#" class="tsd-widget menu" id="tsd-toolbar-menu-trigger" data-toggle="menu" aria-label="Menu"><svg width="16" height="16" viewBox="0 0 16 16" fill="none" aria-hidden="true"><use href="../assets/icons.svg#icon-menu"></use></svg></a></div></header><div class="container container-main"><div class="col-content"><div class="tsd-page-title"><ul class="tsd-breadcrumb" aria-label="Breadcrumb"><li><a href="../modules/FormsAppsTypes.html">FormsAppsTypes</a></li><li><a href="" aria-current="page">FormsAppConfiguration</a></li></ul><h1>Type Alias FormsAppConfiguration&lt;T&gt;</h1></div><div class="tsd-signature"><span class="tsd-kind-type-alias">FormsAppConfiguration</span><span class="tsd-signature-symbol">:</span> <a href="EnvironmentTypes.FormsAppEnvironmentConfiguration.html" class="tsd-signature-type tsd-kind-type-alias">FormsAppEnvironmentConfiguration</a> <span class="tsd-signature-symbol">&amp;</span> <span class="tsd-signature-symbol">{</span><br/>    <span class="tsd-kind-property">cachingStrategies</span><span class="tsd-signature-symbol">?:</span> <a href="FormsAppsTypes._NewFormsApp.html" class="tsd-signature-type tsd-kind-type-alias">_NewFormsApp</a><span class="tsd-signature-symbol">[</span><a href="FormsAppsTypes._NewFormsApp.html#cachingstrategies"><span class="tsd-signature-type">&quot;cachingStrategies&quot;</span></a><span class="tsd-signature-symbol">]</span><span class="tsd-signature-symbol">;</span><br/>    <span class="tsd-kind-property">description</span><span class="tsd-signature-symbol">?:</span> <span class="tsd-signature-type">string</span><span class="tsd-signature-symbol">;</span><br/>    <span class="tsd-kind-property">draftsAreShared</span><span class="tsd-signature-symbol">?:</span> <span class="tsd-signature-type">boolean</span><span class="tsd-signature-symbol">;</span><br/>    <span class="tsd-kind-property">formsAppId</span><span class="tsd-signature-symbol">:</span> <span class="tsd-signature-type">number</span><span class="tsd-signature-symbol">;</span><br/>    <span class="tsd-kind-property">formsHostname</span><span class="tsd-signature-symbol">:</span> <span class="tsd-signature-type">string</span><span class="tsd-signature-symbol">;</span><br/>    <span class="tsd-kind-property">formsOAuthClientId</span><span class="tsd-signature-symbol">?:</span> <span class="tsd-signature-type">string</span> <span class="tsd-signature-symbol">|</span> <span class="tsd-signature-type">null</span><span class="tsd-signature-symbol">;</span><br/>    <span class="tsd-kind-property">googleAnalyticsTagId</span><span class="tsd-signature-symbol">?:</span> <span class="tsd-signature-type">string</span><span class="tsd-signature-symbol">;</span><br/>    <span class="tsd-kind-property">htmlTags</span><span class="tsd-signature-symbol">:</span> <span class="tsd-signature-symbol">{</span> <span class="tsd-kind-property">description</span><span class="tsd-signature-symbol">?:</span> <span class="tsd-signature-type">string</span><span class="tsd-signature-symbol">;</span> <span class="tsd-kind-property">title</span><span class="tsd-signature-symbol">:</span> <span class="tsd-signature-type">string</span> <span class="tsd-signature-symbol">}</span><span class="tsd-signature-symbol">;</span><br/>    <span class="tsd-kind-property">isAppUserMfaRequired</span><span class="tsd-signature-symbol">:</span> <span class="tsd-signature-type">boolean</span><span class="tsd-signature-symbol">;</span><br/>    <span class="tsd-kind-property">isAppUserSignUpEnabled</span><span class="tsd-signature-symbol">:</span> <span class="tsd-signature-type">boolean</span><span class="tsd-signature-symbol">;</span><br/>    <span class="tsd-kind-property">isClientLoggingEnabled</span><span class="tsd-signature-symbol">:</span> <span class="tsd-signature-type">boolean</span><span class="tsd-signature-symbol">;</span><br/>    <span class="tsd-kind-property">isDraftsEnabled</span><span class="tsd-signature-symbol">:</span> <span class="tsd-signature-type">boolean</span><span class="tsd-signature-symbol">;</span><br/>    <span class="tsd-kind-property">isGoogleLoginSupported</span><span class="tsd-signature-symbol">:</span> <span class="tsd-signature-type">boolean</span><span class="tsd-signature-symbol">;</span><br/>    <span class="tsd-kind-property">logoutRedirectUrl</span><span class="tsd-signature-symbol">?:</span> <span class="tsd-signature-type">string</span><span class="tsd-signature-symbol">;</span><br/>    <span class="tsd-kind-property">name</span><span class="tsd-signature-symbol">:</span> <span class="tsd-signature-type">string</span><span class="tsd-signature-symbol">;</span><br/>    <span class="tsd-kind-property">pwaSettings</span><span class="tsd-signature-symbol">?:</span> <a href="FormsAppsTypes.FormsAppPWASettings.html" class="tsd-signature-type tsd-kind-type-alias">FormsAppPWASettings</a> <span class="tsd-signature-symbol">|</span> <span class="tsd-signature-type">null</span><span class="tsd-signature-symbol">;</span><br/>    <span class="tsd-kind-property">samlIdentityProviderName</span><span class="tsd-signature-symbol">?:</span> <span class="tsd-signature-type">string</span> <span class="tsd-signature-symbol">|</span> <span class="tsd-signature-type">null</span><span class="tsd-signature-symbol">;</span><br/>    <span class="tsd-kind-property">styles</span><span class="tsd-signature-symbol">:</span> <a class="tsd-signature-type tsd-kind-type-parameter" href="#t">T</a><span class="tsd-signature-symbol">;</span><br/>    <span class="tsd-kind-property">taskGroupInstances</span><span class="tsd-signature-symbol">?:</span> <a href="../interfaces/ScheduledTasksTypes.TaskGroupInstance.html" class="tsd-signature-type tsd-kind-interface">TaskGroupInstance</a><span class="tsd-signature-symbol">[]</span><span class="tsd-signature-symbol">;</span><br/>    <span class="tsd-kind-property">type</span><span class="tsd-signature-symbol">:</span> <a href="FormsAppsTypes.FormsApp.html" class="tsd-signature-type tsd-kind-type-alias">FormsApp</a><span class="tsd-signature-symbol">[</span><span class="tsd-signature-type">&quot;type&quot;</span><span class="tsd-signature-symbol">]</span><span class="tsd-signature-symbol">;</span><br/>    <span class="tsd-kind-property">volunteers</span><span class="tsd-signature-symbol">:</span><br/>        <span class="tsd-signature-symbol">|</span> <span class="tsd-signature-symbol">{</span><br/>            <span class="tsd-kind-property">categories</span><span class="tsd-signature-symbol">:</span> <a href="FormsAppsTypes.VolunteersFormsApp.html" class="tsd-signature-type tsd-kind-type-alias">VolunteersFormsApp</a><span class="tsd-signature-symbol">[</span><span class="tsd-signature-type">&quot;categories&quot;</span><span class="tsd-signature-symbol">]</span><span class="tsd-signature-symbol">;</span><br/>            <span class="tsd-kind-property">waiverUrl</span><span class="tsd-signature-symbol">:</span> <a href="FormsAppsTypes.VolunteersFormsApp.html" class="tsd-signature-type tsd-kind-type-alias">VolunteersFormsApp</a><span class="tsd-signature-symbol">[</span><span class="tsd-signature-type">&quot;waiverUrl&quot;</span><span class="tsd-signature-symbol">]</span><span class="tsd-signature-symbol">;</span><br/>        <span class="tsd-signature-symbol">}</span><br/>        <span class="tsd-signature-symbol">|</span> <span class="tsd-signature-type">undefined</span><span class="tsd-signature-symbol">;</span><br/><span class="tsd-signature-symbol">}</span></div><section class="tsd-panel"><h4>Type Parameters</h4><ul class="tsd-type-parameter-list"><li><span id="t"><span class="tsd-kind-type-parameter">T</span> <span class="tsd-signature-keyword">extends</span> <a href="EnvironmentTypes.BaseFormsAppEnvironmentStyles.html" class="tsd-signature-type tsd-kind-type-alias">BaseFormsAppEnvironmentStyles</a> = <a href="EnvironmentTypes.BaseFormsAppEnvironmentStyles.html" class="tsd-signature-type tsd-kind-type-alias">BaseFormsAppEnvironmentStyles</a></span></li></ul></section><div class="tsd-type-declaration"><h4>Type declaration</h4><ul class="tsd-parameters"><li class="tsd-parameter"><h5><code class="tsd-tag">Optional</code><span class="tsd-kind-property">cachingStrategies</span><span class="tsd-signature-symbol">?: </span><a href="FormsAppsTypes._NewFormsApp.html" class="tsd-signature-type tsd-kind-type-alias">_NewFormsApp</a><span class="tsd-signature-symbol">[</span><a href="FormsAppsTypes._NewFormsApp.html#cachingstrategies"><span class="tsd-signature-type">&quot;cachingStrategies&quot;</span></a><span class="tsd-signature-symbol">]</span></h5></li><li class="tsd-parameter"><h5><code class="tsd-tag">Optional</code><span class="tsd-kind-property">description</span><span class="tsd-signature-symbol">?: </span><span class="tsd-signature-type">string</span></h5></li><li class="tsd-parameter"><h5><code class="tsd-tag">Optional</code><span class="tsd-kind-property">draftsAreShared</span><span class="tsd-signature-symbol">?: </span><span class="tsd-signature-type">boolean</span></h5></li><li class="tsd-parameter"><h5><span class="tsd-kind-property">formsAppId</span><span class="tsd-signature-symbol">: </span><span class="tsd-signature-type">number</span></h5></li><li class="tsd-parameter"><h5><span class="tsd-kind-property">formsHostname</span><span class="tsd-signature-symbol">: </span><span class="tsd-signature-type">string</span></h5></li><li class="tsd-parameter"><h5><code class="tsd-tag">Optional</code><span class="tsd-kind-property">formsOAuthClientId</span><span class="tsd-signature-symbol">?: </span><span class="tsd-signature-type">string</span> <span class="tsd-signature-symbol">|</span> <span class="tsd-signature-type">null</span></h5></li><li class="tsd-parameter"><h5><code class="tsd-tag">Optional</code><span class="tsd-kind-property">googleAnalyticsTagId</span><span class="tsd-signature-symbol">?: </span><span class="tsd-signature-type">string</span></h5></li><li class="tsd-parameter"><h5><span class="tsd-kind-property">htmlTags</span><span class="tsd-signature-symbol">: </span><span class="tsd-signature-symbol">{</span> <span class="tsd-kind-property">description</span><span class="tsd-signature-symbol">?:</span> <span class="tsd-signature-type">string</span><span class="tsd-signature-symbol">;</span> <span class="tsd-kind-property">title</span><span class="tsd-signature-symbol">:</span> <span class="tsd-signature-type">string</span> <span class="tsd-signature-symbol">}</span></h5></li><li class="tsd-parameter"><h5><span class="tsd-kind-property">isAppUserMfaRequired</span><span class="tsd-signature-symbol">: </span><span class="tsd-signature-type">boolean</span></h5></li><li class="tsd-parameter"><h5><span class="tsd-kind-property">isAppUserSignUpEnabled</span><span class="tsd-signature-symbol">: </span><span class="tsd-signature-type">boolean</span></h5></li><li class="tsd-parameter"><h5><span class="tsd-kind-property">isClientLoggingEnabled</span><span class="tsd-signature-symbol">: </span><span class="tsd-signature-type">boolean</span></h5></li><li class="tsd-parameter"><h5><span class="tsd-kind-property">isDraftsEnabled</span><span class="tsd-signature-symbol">: </span><span class="tsd-signature-type">boolean</span></h5></li><li class="tsd-parameter"><h5><span class="tsd-kind-property">isGoogleLoginSupported</span><span class="tsd-signature-symbol">: </span><span class="tsd-signature-type">boolean</span></h5></li><li class="tsd-parameter"><h5><code class="tsd-tag">Optional</code><span class="tsd-kind-property">logoutRedirectUrl</span><span class="tsd-signature-symbol">?: </span><span class="tsd-signature-type">string</span></h5></li><li class="tsd-parameter"><h5><span class="tsd-kind-property">name</span><span class="tsd-signature-symbol">: </span><span class="tsd-signature-type">string</span></h5></li><li class="tsd-parameter"><h5><code class="tsd-tag">Optional</code><span class="tsd-kind-property">pwaSettings</span><span class="tsd-signature-symbol">?: </span><a href="FormsAppsTypes.FormsAppPWASettings.html" class="tsd-signature-type tsd-kind-type-alias">FormsAppPWASettings</a> <span class="tsd-signature-symbol">|</span> <span class="tsd-signature-type">null</span></h5></li><li class="tsd-parameter"><h5><code class="tsd-tag">Optional</code><span class="tsd-kind-property">samlIdentityProviderName</span><span class="tsd-signature-symbol">?: </span><span class="tsd-signature-type">string</span> <span class="tsd-signature-symbol">|</span> <span class="tsd-signature-type">null</span></h5></li><li class="tsd-parameter"><h5><span class="tsd-kind-property">styles</span><span class="tsd-signature-symbol">: </span><a class="tsd-signature-type tsd-kind-type-parameter" href="#t">T</a></h5></li><li class="tsd-parameter"><h5><code class="tsd-tag">Optional</code><span class="tsd-kind-property">taskGroupInstances</span><span class="tsd-signature-symbol">?: </span><a href="../interfaces/ScheduledTasksTypes.TaskGroupInstance.html" class="tsd-signature-type tsd-kind-interface">TaskGroupInstance</a><span class="tsd-signature-symbol">[]</span></h5></li><li class="tsd-parameter"><h5><span class="tsd-kind-property">type</span><span class="tsd-signature-symbol">: </span><a href="FormsAppsTypes.FormsApp.html" class="tsd-signature-type tsd-kind-type-alias">FormsApp</a><span class="tsd-signature-symbol">[</span><span class="tsd-signature-type">&quot;type&quot;</span><span class="tsd-signature-symbol">]</span></h5><div class="tsd-comment tsd-typography"><p>Type of the forms app.</p>
</div><div class="tsd-comment tsd-typography"></div></li><li class="tsd-parameter"><h5><span class="tsd-kind-property">volunteers</span><span class="tsd-signature-symbol">: </span><br/>    <span class="tsd-signature-symbol">|</span> <span class="tsd-signature-symbol">{</span><br/>        <span class="tsd-kind-property">categories</span><span class="tsd-signature-symbol">:</span> <a href="FormsAppsTypes.VolunteersFormsApp.html" class="tsd-signature-type tsd-kind-type-alias">VolunteersFormsApp</a><span class="tsd-signature-symbol">[</span><span class="tsd-signature-type">&quot;categories&quot;</span><span class="tsd-signature-symbol">]</span><span class="tsd-signature-symbol">;</span><br/>        <span class="tsd-kind-property">waiverUrl</span><span class="tsd-signature-symbol">:</span> <a href="FormsAppsTypes.VolunteersFormsApp.html" class="tsd-signature-type tsd-kind-type-alias">VolunteersFormsApp</a><span class="tsd-signature-symbol">[</span><span class="tsd-signature-type">&quot;waiverUrl&quot;</span><span class="tsd-signature-symbol">]</span><span class="tsd-signature-symbol">;</span><br/>    <span class="tsd-signature-symbol">}</span><br/>    <span class="tsd-signature-symbol">|</span> <span class="tsd-signature-type">undefined</span></h5></li></ul></div><aside class="tsd-sources"><ul><li>Defined in node_modules/@oneblink/types/typescript/formsApps.d.ts:502</li></ul></aside></div><div class="col-sidebar"><div class="page-menu"><div class="tsd-navigation settings"><details class="tsd-accordion"><summary class="tsd-accordion-summary"><svg width="20" height="20" viewBox="0 0 24 24" fill="none" aria-hidden="true"><use href="../assets/icons.svg#icon-chevronDown"></use></svg><h3>Settings</h3></summary><div class="tsd-accordion-details"><div class="tsd-filter-visibility"><span class="settings-label">Member Visibility</span><ul id="tsd-filter-options"><li class="tsd-filter-item"><label class="tsd-filter-input"><input type="checkbox" id="tsd-filter-protected" name="protected"/><svg width="32" height="32" viewBox="0 0 32 32" aria-hidden="true"><rect class="tsd-checkbox-background" width="30" height="30" x="1" y="1" rx="6" fill="none"></rect><path class="tsd-checkbox-checkmark" d="M8.35422 16.8214L13.2143 21.75L24.6458 10.25" stroke="none" stroke-width="3.5" stroke-linejoin="round" fill="none"></path></svg><span>Protected</span></label></li><li class="tsd-filter-item"><label class="tsd-filter-input"><input type="checkbox" id="tsd-filter-inherited" name="inherited" checked/><svg width="32" height="32" viewBox="0 0 32 32" aria-hidden="true"><rect class="tsd-checkbox-background" width="30" height="30" x="1" y="1" rx="6" fill="none"></rect><path class="tsd-checkbox-checkmark" d="M8.35422 16.8214L13.2143 21.75L24.6458 10.25" stroke="none" stroke-width="3.5" stroke-linejoin="round" fill="none"></path></svg><span>Inherited</span></label></li><li class="tsd-filter-item"><label class="tsd-filter-input"><input type="checkbox" id="tsd-filter-external" name="external"/><svg width="32" height="32" viewBox="0 0 32 32" aria-hidden="true"><rect class="tsd-checkbox-background" width="30" height="30" x="1" y="1" rx="6" fill="none"></rect><path class="tsd-checkbox-checkmark" d="M8.35422 16.8214L13.2143 21.75L24.6458 10.25" stroke="none" stroke-width="3.5" stroke-linejoin="round" fill="none"></path></svg><span>External</span></label></li></ul></div><div class="tsd-theme-toggle"><label class="settings-label" for="tsd-theme">Theme</label><select id="tsd-theme"><option value="os">OS</option><option value="light">Light</option><option value="dark">Dark</option></select></div></div></details></div></div><div class="site-menu"><nav class="tsd-navigation"><a href="../modules.html">@oneblink/sdk-core</a><ul class="tsd-small-nested-navigation" id="tsd-nav-container"><li>Loading...</li></ul></nav></div></div></div><footer></footer><div class="overlay"></div></body></html>
