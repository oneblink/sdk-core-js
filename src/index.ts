export * as formElementsService from './formElementsService'
export * as conditionalLogicService from './conditionalLogicService'
import replaceCustomValues, {
  getElementSubmissionValue,
} from './replaceCustomValues'
export * as typeCastService from './typeCastService'
export * as userService from './userService'
export * as abnService from './abnService'
export { validatePaymentAmount } from './paymentAmountValidation'
export { getContentDisposition } from './fileUploadService'

export { replaceCustomValues, getElementSubmissionValue }
