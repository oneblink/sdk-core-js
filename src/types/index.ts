export type FormElementsCtrl = {
  model: Record<string, unknown>
  flattenedElements: import('@oneblink/types').FormTypes.FormElement[]
  parentFormElementsCtrl?: FormElementsCtrl
}

export type FormElementsConditionallyShown = Record<
  string,
  FormElementConditionallyShown
>

export type FormElementConditionallyShown =
  | undefined
  | {
      type: 'formElement'
      isHidden: boolean
      options?: import('@oneblink/types').FormTypes.ChoiceElementOption[]
    }
  | {
      type: 'formElements'
      isHidden: boolean
      formElements: FormElementsConditionallyShown | undefined
    }
  | {
      type: 'repeatableSet'
      isHidden: boolean
      entries: Record<string, FormElementsConditionallyShown | undefined>
    }

