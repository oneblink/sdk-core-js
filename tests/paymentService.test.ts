import { SubmissionTypes } from '@oneblink/types'
import { paymentService } from '../src'

describe('getDisplayDetailsFromFormSubmissionPayment', () => {
  const formatters = {
    formatCurrency: (amount: number) => `$${amount.toFixed(2)}`,
    formatDateTime: () => '11/09/2024 12:00:00 PM',
    formatDate: () => '11/09/2024',
  }

  describe('NSW Gov Pay', () => {
    const formSubmissionPayment: SubmissionTypes.FormSubmissionPayment = {
      type: 'NSW_GOV_PAY',
      id: '1',
      createdAt: '2024-09-11T12:00:00.000Z',
      formId: 1,
      status: 'SUCCEEDED',
      submissionId: '111111111111',
      updatedAt: '2024-09-11T12:00:00.000Z',
      paymentTransaction: {
        agencyCompletionPayment: {
          paymentCompletionReference: '111111111111',
          paymentReference: '222222222222',
          bankReference: '333333333333',
          paymentMethod: 'CARD',
          amount: 123.45,
          surcharge: 12.34,
          surchargeGst: 2.34,
          agencyTransactionId: 'agencyTransactionId',
          bPay: {
            billerCode: 'billerCode',
            crn: 'crn',
            processingDate: '2024-09-11T12:00:00.000Z',
          },
          card: {
            last4Digits: '4242',
            cardPresent: true,
            cardType: 'VISA',
          },
        },
        integrationPrimaryAgencyId: '1',
        nswGovPayPaymentReference: 'nswGovPayPaymentReference',
        redirectUrl: 'redirectUrl',
      },
    }

    it('gets the correct details', () => {
      const details = paymentService.getDisplayDetailsFromFormSubmissionPayment(
        formSubmissionPayment,
        formatters,
      )
      expect(details).toMatchSnapshot()
    })

    it('using BPAY', () => {
      const details = paymentService.getDisplayDetailsFromFormSubmissionPayment(
        {
          ...formSubmissionPayment,
          paymentTransaction: {
            ...formSubmissionPayment.paymentTransaction,
            // @ts-expect-error Wrong
            agencyCompletionPayment: {
              ...formSubmissionPayment.paymentTransaction
                .agencyCompletionPayment,
              paymentMethod: 'BPAY',
            },
          },
        },
        formatters,
      )

      expect(details.find((d) => d.key === 'billerCode')?.value).toBe(
        'billerCode',
      )
    })
  })
  describe('Bpoint', () => {
    const formSubmissionPayment: SubmissionTypes.FormSubmissionPayment = {
      type: 'BPOINT',
      id: '1',
      createdAt: '2024-09-11T12:00:00.000Z',
      formId: 1,
      status: 'SUCCEEDED',
      submissionId: '111111111111',
      updatedAt: '2024-09-11T12:00:00.000Z',
      paymentTransaction: {
        Amount: 12345,
        AmountSurcharge: 1234,
        Action: 'PAYMENT',
        AmountOriginal: 12345,
        ReceiptNumber: 'receiptNumber',
        BillerCode: 'billerCode',
        Crn1: 'crn1',
        Crn2: 'crn2',
        Crn3: 'crn3',
        // @ts-expect-error incomplete because type is large
        CardDetails: {
          MaskedCardNumber: 'creditCardMask',
        },
        ProcessedDateTime: '2024-09-11T12:00:00.000Z',
      },
    }
    it('gets the correct details', () => {
      const details = paymentService.getDisplayDetailsFromFormSubmissionPayment(
        formSubmissionPayment,
        formatters,
      )
      expect(details).toMatchSnapshot()
    })
  })
  describe('CP Pay', () => {
    const formSubmissionPaymentv1: SubmissionTypes.FormSubmissionPayment = {
      type: 'CP_PAY',
      id: '1',
      createdAt: '2024-09-11T12:00:00.000Z',
      formId: 1,
      status: 'SUCCEEDED',
      submissionId: '111111111111',
      updatedAt: '2024-09-11T12:00:00.000Z',
      // @ts-expect-error incomplete because type is large
      paymentTransaction: {
        cpPayVersion: 'v1',
        transactionId: '123',
        orderNumber: 'orderNumber',
        paymentTypeId: 1,
        lastFour: '1234',
        amount: 123.45,
        createdAt: '2024-09-11T12:00:00.000Z',
      },
    }
    const formSubmissionPaymentv2: SubmissionTypes.FormSubmissionPayment = {
      type: 'CP_PAY',
      id: '1',
      createdAt: '2024-09-11T12:00:00.000Z',
      formId: 1,
      status: 'SUCCEEDED',
      submissionId: '111111111111',
      updatedAt: '2024-09-11T12:00:00.000Z',
      // @ts-expect-error incomplete because type is large
      paymentTransaction: {
        cpPayVersion: 'v2',
        result: {
          id: '123',
          externalReferenceId: 'externalReferenceId',
          paymentType: 'CreditDebitCard',
          lastFour: '1234',
          amount: 123.45,
          createdOnUtc: '2024-09-11T12:00:00.000Z',
        },
      },
    }

    it('gets the correct details - v1', () => {
      const details = paymentService.getDisplayDetailsFromFormSubmissionPayment(
        formSubmissionPaymentv1,
        formatters,
      )
      expect(details).toMatchSnapshot()
    })

    it('gets the correct details - v2', () => {
      const details = paymentService.getDisplayDetailsFromFormSubmissionPayment(
        formSubmissionPaymentv2,
        formatters,
      )
      expect(details).toMatchSnapshot()
    })
  })

  describe('Westpac', () => {
    const formSubmissionPayment: SubmissionTypes.FormSubmissionPayment = {
      type: 'WESTPAC_QUICK_STREAM',
      id: '1',
      createdAt: '2024-09-11T12:00:00.000Z',
      formId: 1,
      status: 'SUCCEEDED',
      submissionId: '111111111111',
      updatedAt: '2024-09-11T12:00:00.000Z',
      // @ts-expect-error unfinished because type is large
      paymentTransaction: {
        receiptNumber: 'receiptNumber',
        paymentReferenceNumber: 'paymentReferenceNumber',
        customerReferenceNumber: 'customerReferenceNumber',
        totalAmount: {
          amount: 123.45,
          displayAmount: '123.45',
          currency: 'AUD',
        },
        surchargeAmount: {
          amount: 12.34,
          currency: 'AUD',
          displayAmount: '12.34',
        },
        settlementDate: '2024-09-11T12:00:00.000Z',
      },
    }

    it('gets the correct details', () => {
      const details = paymentService.getDisplayDetailsFromFormSubmissionPayment(
        formSubmissionPayment,
        formatters,
      )
      expect(details).toMatchSnapshot()
    })
  })
})
