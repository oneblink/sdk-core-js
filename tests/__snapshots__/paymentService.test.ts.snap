// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getDisplayDetailsFromFormSubmissionPayment Bpoint gets the correct details 1`] = `
[
  {
    "key": "receiptNumber",
    "label": "Receipt Number",
    "value": "receiptNumber",
  },
  {
    "key": "crn1",
    "label": "CRN 1",
    "value": "crn1",
  },
  {
    "key": "crn2",
    "label": "CRN 2",
    "value": "crn2",
  },
  {
    "key": "crn3",
    "label": "CRN 3",
    "value": "crn3",
  },
  {
    "key": "billerCode",
    "label": "Biller Code",
    "value": "billerCode",
  },
  {
    "key": "creditCardMask",
    "label": "Card Number",
    "value": "creditCardMask",
  },
  {
    "key": "amount",
    "label": "Amount",
    "value": "$123.45",
  },
  {
    "key": "surchargeAmount",
    "label": "Surcharge Amount",
    "value": "$12.34",
  },
  {
    "key": "processedDateTime",
    "label": "Processed Date Time",
    "value": "11/09/2024 12:00:00 PM",
  },
]
`;

exports[`getDisplayDetailsFromFormSubmissionPayment CP Pay gets the correct details - v1 1`] = `
[
  {
    "key": "transactionId",
    "label": "Transaction Id",
    "value": "123",
  },
  {
    "key": "orderNumber",
    "label": "Order Number",
    "value": "orderNumber",
  },
  {
    "key": "paymentType",
    "label": "Payment Type",
    "value": "Credit/Debit Card",
  },
  {
    "key": "creditCardMask",
    "label": "Card Number",
    "value": "xxxx xxxx xxxx 1234",
  },
  {
    "key": "amount",
    "label": "Amount",
    "value": "$123.45",
  },
  {
    "key": "createdDateTime",
    "label": "Created At",
    "value": "11/09/2024 12:00:00 PM",
  },
]
`;

exports[`getDisplayDetailsFromFormSubmissionPayment CP Pay gets the correct details - v2 1`] = `
[
  {
    "key": "transactionId",
    "label": "Transaction Id",
    "value": "123",
  },
  {
    "key": "orderNumber",
    "label": "Order Number",
    "value": "externalReferenceId",
  },
  {
    "key": "paymentType",
    "label": "Payment Type",
    "value": "CreditDebitCard",
  },
  {
    "key": "creditCardMask",
    "label": "Card Number",
    "value": "xxxx xxxx xxxx 1234",
  },
  {
    "key": "amount",
    "label": "Amount",
    "value": "$123.45",
  },
  {
    "key": "createdDateTime",
    "label": "Created At",
    "value": "11/09/2024 12:00:00 PM",
  },
]
`;

exports[`getDisplayDetailsFromFormSubmissionPayment NSW Gov Pay gets the correct details 1`] = `
[
  {
    "key": "completionReference",
    "label": "Completion Reference",
    "value": "111111111111",
  },
  {
    "key": "paymentReference",
    "label": "Payment Reference",
    "value": "222222222222",
  },
  {
    "key": "bankReference",
    "label": "Bank Reference",
    "value": "333333333333",
  },
  {
    "key": "paymentMethod",
    "label": "Payment Method",
    "value": "CARD",
  },
  {
    "key": "creditCardMask",
    "label": "Card Number",
    "value": "xxxx xxxx xxxx 4242",
  },
  {
    "key": "amount",
    "label": "Amount",
    "value": "$123.45",
  },
  {
    "key": "surchargeAmount",
    "label": "Surcharge Amount",
    "value": "$12.34",
  },
  {
    "key": "surchargeGST",
    "label": "Surcharge GST",
    "value": "$2.34",
  },
  {
    "key": "createdDateTime",
    "label": "Created Date Time",
    "value": "11/09/2024 12:00:00 PM",
  },
]
`;

exports[`getDisplayDetailsFromFormSubmissionPayment Westpac gets the correct details 1`] = `
[
  {
    "key": "receiptNumber",
    "label": "Receipt Number",
    "value": "receiptNumber",
  },
  {
    "key": "paymentReferenceNumber",
    "label": "Payment Reference",
    "value": "paymentReferenceNumber",
  },
  {
    "key": "customerReferenceNumber",
    "label": "Customer Reference Number",
    "value": "customerReferenceNumber",
  },
  {
    "key": "amount",
    "label": "Amount",
    "value": "$123.45",
  },
  {
    "key": "surchargeAmount",
    "label": "Surcharge Amount",
    "value": "$12.34",
  },
  {
    "key": "settlementDate",
    "label": "Settlement Date",
    "value": "11/09/2024 12:00:00 PM",
  },
]
`;
